plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -903,11 +903,11 @@ end
 
 function ListMenu:_updateItemsBuildUI()
     -- Build our list
-    table.insert(self.item_group, LineWidget:new{
-                    dimen = Geom:new{ w = self.width, h = Size.line.thin },
-                    background = Blitbuffer.COLOR_DARK_GRAY,
-                    style = "solid",
-                })
+    local line_widget = LineWidget:new{
+        dimen = Geom:new{ w = self.width or self.screen_w, h = Size.line.thin },
```

Comment:
This was the primary reason for not seeing it (no self.width provided when called from History I guess), right? and the rest is just refactoring?

Reply:
Yes, caused with my https://github.com/koreader/koreader/pull/10244/commits/48e63c01c4f7aed466b02d18cba7c767209bf0ce.
Is it a bad refactoring, should be reverted?

Reply:
> Is it a bad refactoring, should be reverted?

No no, no problem with the refactoring and using a single LineWidget (I think we do that, reusing a single simple widget, in many places - I just checked that LineWidget does not update/use any dimen.x/y, so I guess that is fine).

I thought the `self.width or self.screen_w` might be not the best as there, as it's solving it that these precise places, and there could be other places that would use `self.width` - but looking precisely in ListMenu and MosaicMenu, I don't find any other use than these LineWidget, so ok.